src/Calculator/Views/GraphingCalculator/EquationInputArea.xaml.cpp

```diff
@@ -67,8 +69,16 @@ void EquationInputArea::AddNewEquation()
         Equations->GetAt(Equations->Size - 1)->IsLastItemInList = false;
     }
 
-    m_lastLineColorIndex = (m_lastLineColorIndex + 1) % AvailableColors->Size;
-    auto eq = ref new EquationViewModel(ref new Equation(), ++m_lastFunctionLabelIndex, AvailableColors->GetAt(m_lastLineColorIndex)->Color);
+    // Cap equations at number of available colors
+    if (Equations->Size >= m_AvailableColors->Size)
```

Comment:
Should we?

What are we trying to solve?

If we don't want 2 equations to not use the same color, this won't prevent it (if you remove an equation and add a new one, it will probably reuse the same color than an existing one). IMO, we should not limit the number of equations because of this reason

Reply:
Especially in High Contrast (only 4 colors/equations)

Reply:
Will defer to @grochocki as this was a change he wanted. I don't think the goal is to have unique colors for each equation. @grochocki this is the change to limit the number of equations to 14.

Reply:
But it was not intended to limit to only 4 in high contrast, will fix that